src/main/scala/io/viash/functionality/dependencies/Repository.scala

```diff
@@ -150,11 +151,32 @@ case class GithubRepository(
     copy(localPath = temporaryFolder.toString)
   }
 
+  def hasBranch(name: String, cwd: Option[File]): Boolean = {
+    val out = Exec.runCatch(
+      List("git", "show-ref", "--verify", "--quiet", s"refs/heads/$name"),
+      cwd = cwd
+    )
+    out.exitValue == 0
+  }
+
+  def hasTag(name: String, cwd: Option[File]): Boolean = {
+    val out = Exec.runCatch(
+      List("git", "show-ref", "--verify", "--quiet", s"refs/tags/$name"),
+      cwd = cwd
+    )
+    out.exitValue == 0
+  }
+
   // Checkout of files from already cloned repository. Limit file checkout to the path that was specified
   def checkout(): GithubRepository = {
-    val checkoutName = "origin/" + tag.getOrElse("HEAD")
     val pathStr = path.getOrElse(".")
     val cwd = Some(Paths.get(localPath).toFile)
+    val checkoutName = tag match {
+      case Some(name) if hasBranch(name, cwd) => s"origin/$name"
```

Comment:
What if the remote is not called origin?

Reply:
This code assumes we *just* did a checkout with the default naming strategy for remote repositories. For now, I'm not sure what could cause the naming to be different.